tests/test_interfaces.py

```diff
@@ -0,0 +1,595 @@
+
+        with self.assertRaises(interfaces.InterfaceCantBeFoundError):
+            self.network_manager.get_interface(True)
+
+    def test_get_interface_has_interface_interface(self):
+        """
+        Tests get_interface method when interface with specified mode
+        can be found
+        """
+
+        interface_name = "wlan0"
+        interface_object = "Card Object"
+        adapter = interfaces.NetworkAdapter(interface_name, interface_object)
+        adapter.has_ap_mode = True
+        adapter.has_monitor_mode = True
+        self.network_manager._name_to_object[interface_name] = adapter
+
+        expected = interface_name
+        actual = self.network_manager.get_interface(True, True)
+
+        self.assertEqual(expected, actual)
+
+    def test_get_interface_automatically_no_interface_error(self):
+        """
+        Tests get_interface_automatically method when no interface
+        is found
+        """
+
+        with self.assertRaises(interfaces.InterfaceCantBeFoundError):
+            self.network_manager.get_interface_automatically()
+
+    def test_get_interface_automatically_monitor_only_interface_error(self):
+        """
+        Tests get_interface_automatically method when only monitor
+        interface is found
+        """
+
+        interface_name_0 = "wlan0"
+        interface_name_1 = "wlan1"
+        interface_object = "Card Object"
+        adapter_0 = interfaces.NetworkAdapter(interface_name_0, interface_object)
+        adapter_1 = interfaces.NetworkAdapter(interface_name_0, interface_object)
+        adapter_0.has_monitor_mode = True
+        adapter_1.has_monitor_mode = True
+        self.network_manager._name_to_object[interface_name_0] = adapter_0
+        self.network_manager._name_to_object[interface_name_1] = adapter_1
+
+        with self.assertRaises(interfaces.InterfaceCantBeFoundError):
+            self.network_manager.get_interface_automatically()
+
+    def test_get_interface_automatically_ap_only_interface_error(self):
+        """
+        Tests get_interface_automatically method when only AP interface
+        is found
+        """
+
+        interface_name_0 = "wlan0"
+        interface_name_1 = "wlan1"
+        interface_object = "Card Object"
+        adapter_0 = interfaces.NetworkAdapter(interface_name_0, interface_object)
+        adapter_1 = interfaces.NetworkAdapter(interface_name_0, interface_object)
+        adapter_0.has_ap_mode = True
+        adapter_1.has_ap_mode = True
+        self.network_manager._name_to_object[interface_name_0] = adapter_0
+        self.network_manager._name_to_object[interface_name_1] = adapter_1
+
+        with self.assertRaises(interfaces.InterfaceCantBeFoundError):
+            self.network_manager.get_interface_automatically()
+
+    def test_get_interface_automatically_has_interface_interfaces(self):
+        """
+        Tests get_interface_automatically method when two interfaces
+        can be found
+        """
```

Comment:
This is one of the most important tests. But you only examine one simple case.

You should examine all possible weird corner cases.